src/cryptonote_core/miner.h

```diff
@@ -119,8 +139,39 @@ namespace cryptonote
     bool m_do_print_hashrate;
     bool m_do_mining;
 
-  };
-}
-
+    // background mining stuffs ..
 
+    static const int CPU_DATA_OVERALL_TIMES_COUNT = 4;
+    static const int CPU_DATA_OVERALL_USER_INDEX = 0;
+    static const int CPU_DATA_OVERALL_NICE_INDEX = 1;
+    static const int CPU_DATA_OVERALL_SYSTEM_INDEX = 2;
+    static const int CPU_DATA_OVERALL_IDLE_INDEX = 3;
+    //const int CPU_DATA_PROCESS_TIMES_COUNT;
+    struct cpu_data
+    {
+      uint64_t update_time; // set to 0 to update
+      uint64_t overall_cpu_times[CPU_DATA_OVERALL_TIMES_COUNT];
+      uint64_t process_cpu_time;
+    };
 
+    bool m_is_background_mining_enabled;
+    epee::math_helper::once_a_time_seconds<BACKGROUND_MINING_DEFAULT_MIN_IDLE_INTERVAL_IN_SECONDS> m_decide_start_interval;
+    epee::math_helper::once_a_time_seconds<10> m_decide_stop_interval; /* decision to stop should be quicker than start */
+    //boost::circular_buffer<cpu_stat_data> m_cpu_stats(2);
+    cpu_data m_cpu_data_start, m_cpu_data_end;
```

Comment:
Only `m_cpu_data_start` is needed as a member variable. `m_cpu_data_end` can be created on the stack when polling, and then copied into `m_cpu_data_start`.